monitor/xfeemngr/contract/bound.go

```diff
@@ -0,0 +1,130 @@
+package contract
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+
+	"github.com/omni-network/omni/contracts/bindings"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/ethclient"
+	"github.com/omni-network/omni/lib/ethclient/ethbackend"
+	"github.com/omni-network/omni/lib/netconf"
+	"github.com/omni-network/omni/lib/xchain"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+type BoundFeeOracleV1 struct {
+	owner   common.Address        // eoa owner of the FeeOracleV1 dcontract
+	addr    common.Address        // address of the FeeOracle oracle contract addrss
+	backend *ethbackend.Backend   // ethbackend initialized with owner pk
+	bound   *bindings.FeeOracleV1 // bound FeeOracleV1 contract
+	_txOpts *bind.TransactOpts    // transaction opts - use txOpts(ctx) instead
```

Comment:
Was gonna ask on suggested pattern. I put underscore,  because i have a helper `contract.txOpts(ctx)` that basically clones _txopts but adds a new context

Reply:
We don't need the `_`, IMHO. Add a good documentation.

Reply:
Yeah I ended up doing `txOptsWithCtx`